holoviews/core/data/xarray.py

```diff
@@ -422,6 +422,10 @@ def values(cls, dataset, dim, expanded=True, flat=True, compute=True, keep_index
             return data.T.flatten() if flat and not keep_index else data
         elif expanded:
             data = cls.coords(dataset, dim.name, expanded=True)
+            if keep_index:
+                da = dataset.data[dataset.vdims[0].name].copy().rename(dim.name)
+                da.data[:] = data
+                return da
```

Comment:
Need to find a better solution here. This basically ensures that `keep_index` + `expanded` returns a 2D coordinate array even for a rectilinear (i.e. 1D coordinate) array. This implementation does not handle:

- packed vdims
- dask backed value arrays